src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -441,14 +523,11 @@ ASTPtr MutationsInterpreter::prepare(bool dry_run)
     NamesAndTypesList all_columns = columns_desc.getAllPhysical();
 
     /// Add _row_exists column if it is physically present in the part
-    if (auto part_storage = dynamic_pointer_cast<DB::StorageFromMergeTreeDataPart>(storage))
-    {
-        if (part_storage->hasLightweightDeletedMask())
+    if (source.hasLightweightDeleteMask())
             all_columns.push_back({LightweightDeleteDescription::FILTER_COLUMN});
```

Comment:
You might want to fix indentation of this line :)

Reply:
oops